lib/index.js

```diff
@@ -34,10 +34,13 @@ const glob = require('glob')
 const globify = pattern => pattern.split('\\').join('/')
 
 const readOutOfTreeIgnoreFiles = (root, rel, result = '') => {
-  for (const file of ['.gitignore', '.npmignore']) {
+  for (const file of ['.npmignore', '.gitignore']) {
     try {
       const ignoreContent = fs.readFileSync(path.join(root, file), { encoding: 'utf8' })
       result += ignoreContent + '\n'
+      // break the loop immediately after concatting, this allows us to prioritize the
+      // .npmignore and discard the .gitignore if one exists
+      break
     } catch (err) {
```

Comment:
I reckon that, if I have a `npmignore` and `gitignore` at `workspaces/foo` but only a `gitignore` in workspaces (or any folder in between) or at the root level, I will endup with a result:
```
content of npmignore from foo
content of gitignore from workspace
content of gitignore from root
```

It seems odd to me to mix the npmignore and gitignore (ofc, if that the expected behavior, all good 👍 )

Reply:
that's the expected behavior, if a directory does not contain a `.npmignore` but does have a `.gitignore` then we respect the `.gitignore`

Reply:
👍 if that's the expected behaviour, then all's well.
I think you still need to reverse the result concatenation because you're crawling up the tree, but I might be mistaken.
```suggestion
  for (const file of ['.npmignore', '.gitignore']) {
    try {
      const ignoreContent = fs.readFileSync(path.join(root, file), { encoding: 'utf8' })
      result = ignoreContent + '\n' + result
      // break the loop immediately after concatting, this allows us to prioritize the
      // .npmignore and discard the .gitignore if one exists
      break
    } catch (err) {
```

Reply:
we're crawling from the root deeper toward the workspace, you can see when we call `readOutOfTreeIgnoreFiles` here: https://github.com/npm/npm-packlist/blob/main/lib/index.js#L166 we're passing it `opt.prefix` which is the `/root`, and `relpath` to carry with your examples in #108 would be the string `packages`

the loop runs with `root`, then appends the next segment of the relpath (`packages`) and runs again, concatting this result to the end, this gives us the expected order of rules